source/CommandLine/CommandOutputProviderBase.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Octopus.CommandLine.OptionParsing;
+using Serilog;
+using Serilog.Events;
+
+namespace Octopus.CommandLine
+{
+    public abstract class CommandOutputProviderBase : ICommandOutputProvider
+    {
+        readonly ILogger logger;
+
+        protected CommandOutputProviderBase(ILogger logger)
+        {
+            this.logger = logger;
+            PrintMessages = true; // unless told otherwise
+        }
+
+        public bool PrintMessages { get; set; }
+
+        public void PrintHeader()
+        {
+            if (PrintMessages)
+            {
+                logger.Information($"{GetAppName()}, version {GetAppVersion()}");
+                logger.Information(string.Empty);
+            }
+        }
+
+        protected abstract string GetAppName();
```

Comment:
We used to hardcode the app name. Now we push it down for implementers to provide it (and we provide a nice default via `DefaultCommandOutputProvider`)